features/steps.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+from contextlib import closing
+import datetime
+import os
+from journal import INSERT_ENTRY
+from journal import connect_db
+from journal import DB_SCHEMA
+from lettuce import world
+from lettuce import step
+from lettuce import before
+from lettuce import after
+
+TEST_DSN = 'dbname=test_learning_journal user=nbeck'
+
+
+def init_db(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute(DB_SCHEMA)
+        db.commit()
+
+
+def empty_db(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute("DROP TABLE entries")
+        db.commit()
+
+
+def clear_entries(setup):
+    with closing(connect_db(setup)) as db:
+        db.cursor().execute("DELETE FROM entries")
+        db.commit()
+
+
+@before.all
+def db():
+    setup = {'db': TEST_DSN}
+    init_db(setup)
+    world.setup = setup
+
+
+@before.all
+def app():
+    from journal import main
+    from webtest import TestApp
+    os.environ['DATABASE_URL'] = TEST_DSN
+    app = main()
+    world.app = TestApp(app)
```

Comment:
running this function `@before.all` means that the same exact test app is shared among all the tests.  this can lead to problems in test isolation, which is not good (trust me, I just spent three days fighting such a problem).  The same can be said for the functions below where you login and create entries.  They should happen before and after each scenario, not just once per run.